src/addons/Display/backend.py

```diff
@@ -0,0 +1,228 @@
+                raise
+            else:
+                pass
+
+    if backend_str == "qt-pyside" or \
+            (backend_str is None and not HAVE_BACKEND):
+        try:
+            from PySide import QtCore, QtGui, QtOpenGL
+            QtWidgets = QtGui
+
+            HAVE_BACKEND = True
+            HAVE_PYSIDE = True
+            BACKEND_MODULE = 'qt-pyside'
+            log.info("backend loaded: {0}".format(BACKEND_MODULE))
+            return BACKEND_MODULE
+
+        except ImportError:
+            if backend_str is not None:
+                log.exception("{0} could not be loaded".format(backend_str))
+                raise
+            else:
+                pass
+
+    if backend_str == "wx" or \
+            (backend_str is None and not HAVE_BACKEND):
+        try:
+            import wx
+
+            warnings.warn("wx for pythonocc development is in maintanence mode"
+                          "use Qt ( PqyQt5, PyQt4 or PySide ) for new projects",
+                          DeprecationWarning, level=2)
+            HAVE_BACKEND = True
+            HAVE_WX = True
+            BACKEND_MODULE = 'wx'
+            log.info("backend loaded: {0}".format(BACKEND_MODULE))
+            return BACKEND_MODULE
+
+        except:
+            if backend_str is not None:
+                log.exception("{0} could not be loaded".format(backend_str))
+                raise
+            else:
+                pass
+
+    if not HAVE_BACKEND:
+        raise ImportError(
+                "No compliant GUI library could be imported.\n "
+                "Either PyQt5, PyQt4, PySide, or wxPython is required")
+
+
+def get_qt_modules():
+    """
+
+    Returns
+    -------
+    tuple : ( QtCore, QtGui, QtWidgets, QtOpenGL )
+        here QtWidgets shadows QtGui when a PyQt4 or PySide module is loaded
+        this is the most coherent way to get PyQt5 compliant code
+
+    Raises
+    ------
+
+    ValueError
+        when no Qt backend has been yet loaded
+        informs the user to call `get_backend` or that no Qt python module
+        ( PyQt5, PyQt4 or PySide ) is found
+
+    """
+
+    global HAVE_BACKEND, HAVE_PYQT5, HAVE_PYQT4, HAVE_PYSIDE, HAVE_WX, QtCore, \
+        QtGui, QtWidgets, QtOpenGL
+
+    if not HAVE_BACKEND:
+        raise ValueError("no backend has been imported yet with "
+                         "``get_backend``... ")
+
+    if HAVE_PYQT5 or HAVE_PYQT4 or HAVE_PYSIDE:
+        return QtCore, QtGui, QtWidgets, QtOpenGL
+    elif HAVE_WX:
+        raise ValueError("the Wx backend is already loaded")
+    else:
+        msg = ("no Qt backend is loaded, hence cannot return any modules\n"
+               "either you havent got PyQt5, PyQt4 or PySide installed\n"
+               "or you havent yet loaded a backend with the "
+               "`OCC.Display.backend.get_backend` function")
+
+        raise ValueError(msg)
+
+
+def have_pyqt5():
+    global HAVE_PYQT5
```

Comment:
shouldn't we make a check here, i.e. try to import Qt5 modules. Indeed if no display has been inited, these functions will always return False.